crates/assessor/src/lib.rs

```diff
@@ -66,6 +66,29 @@ impl Fulfillment {
     }
 }
 
+/// Computes a commitment hash.
+///
+/// The commitment hash is computed as the keccak256 hash of the concatenation of the following:
+/// - The index of the request in the batch.
+/// - The request ID.
+/// - The request digest.
+/// - The claim digest.
+pub fn commit(
```

Comment:
We should use a struct here and domain separation (e.g. EIP-712) instead of just concatting these bytes.

Reply:
Gas released issue fixed in [cdaceae](https://github.com/boundless-xyz/boundless/pull/441/commits/cdaceae45cb35924c6036775741b3170cae6fa3a)